### ISSUE TITLE
allow-redefinition should not be enabled by default

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#609]
allow redefinition is basically the same as shadowing, which is cringe imo

**Comment by KotlinIsland** [comment]
I don't agree at this point. I don't view it as shadowing. I can understand that it can be accidentally used, but the point of it is that it allows you to write more natural feeling python.
